godot-google-play-billing/src/main/java/org/godotengine/godot/plugin/googleplaybilling/GodotGooglePlayBilling.java

```diff
@@ -119,37 +132,42 @@ public void onQueryPurchasesResponse(BillingResult billingResult,
 		});
 	}
 
+	@UsedByGodot
 	public void querySkuDetails(final String[] list, String type) {
-		List<String> skuList = Arrays.asList(list);
+		ArrayList<QueryProductDetailsParams.Product> products = new ArrayList<>();
+
+		for (String productId : list) {
+			products.add(QueryProductDetailsParams.Product.newBuilder().setProductId(productId).setProductType(type).build());
+		}
 
-		SkuDetailsParams.Builder params = SkuDetailsParams.newBuilder()
-												  .setSkusList(skuList)
-												  .setType(type);
+		QueryProductDetailsParams params = QueryProductDetailsParams.newBuilder()
+												   .setProductList(products)
+												   .build();
 
-		billingClient.querySkuDetailsAsync(params.build(), new SkuDetailsResponseListener() {
+		billingClient.queryProductDetailsAsync(params, new ProductDetailsResponseListener() {
 			@Override
-			public void onSkuDetailsResponse(BillingResult billingResult,
-					List<SkuDetails> skuDetailsList) {
+			public void onProductDetailsResponse(@NonNull BillingResult billingResult, @NonNull List<ProductDetails> list) {
 				if (billingResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
-					for (SkuDetails skuDetails : skuDetailsList) {
-						skuDetailsCache.put(skuDetails.getSku(), skuDetails);
+					for (ProductDetails productDetails : list) {
+						productDetailsCache.put(productDetails.getProductId(), productDetails);
 					}
-					emitSignal("sku_details_query_completed", (Object)GooglePlayBillingUtils.convertSkuDetailsListToDictionaryObjectArray(skuDetailsList));
+					emitSignal("product_details_query_completed", (Object)GooglePlayBillingUtils.convertProductDetailsListToDictionaryObjectArray(list));
```

Comment:
I don't understand why this is being cast to `Object` -- isn't it already an Object?  (It's actually Object[], but that is, itself, an object).